File: dm-test/src/test/java/ro/johann/dm/test/api/service/decision/transfer/CriteriaOutput.java
package ro.johann.dm.test.api.service.decision.transfer;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import static lombok.AccessLevel.PRIVATE;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = PRIVATE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class CriteriaOutput {
  String id;
  String name;
  int weight;
  String unitOfMeasure;
  String type;
}

File: dm-decision-service/src/main/kotlin/ro/johann/dm/decision/api/transfer/CriteriaOutput.kt
package ro.johann.dm.decision.api.transfer

import ro.johann.dm.decision.domain.Criteria
import java.util.UUID

data class CriteriaOutput(
  val id: UUID,
  val name: String,
  val weight: Int,
  val unitOfMeasure: String?,
) {
  constructor(criteria: Criteria) : this(
    id = criteria.id,
    name = criteria.name,
    weight = criteria.weight,
    unitOfMeasure = criteria.unitOfMeasure
  )
}